src/tictactoe/game_menu.clj

```diff
@@ -0,0 +1,59 @@
+(ns tictactoe.game-menu
+  (:require [clojure.string :as string]
+            [tictactoe.user-interface :as ui]
+            [tictactoe.game-messages :as messages]
+            [tictactoe.menu-validation :as validator]
+            [tictactoe.marker-validation :as marker-validator]))
+
+(defn welcome-message []
+  (ui/print-message (messages/welcome)))
+
+(defn pick-game [game-types] 
+  (ui/print-message (messages/game-type))
+  (ui/loop-and-print game-types)
+  (-> 
+    (ui/get-input)
+    (keyword)
+    (validator/input-validation-loop game-types)
+    (keyword)))
+
+(defn pick-board-size [board-sizes] 
+  (ui/print-message (messages/board-type))
+  (ui/loop-and-print board-sizes)
+  (-> 
+    (ui/get-input)
+    (keyword)
+    (validator/input-validation-loop board-sizes)
+    (keyword)))
+
+(defn pick-player-one-marker []
+  (ui/print-message (messages/player-one-marker))
+  (-> 
+    (ui/get-input)
+    (marker-validator/marker-validation-loop)
+    (string/upper-case)))
+
+(defn pick-player-two-marker [player1-marker]
```

Comment:
same for the `pick-player-marker`. Can you use an optional argument and abstract this function?